File: 389-find-the-difference/find-the-difference.py
class Solution(object):
    def findTheDifference(self, s, t):
        
        maps = {}
        mapt = {}
        for ele in s:
            if ele in maps:
                maps[ele] += 1
            else:
                maps[ele] = 1

        for ele in t:
            if ele in mapt:
                mapt[ele] += 1
            else:
                mapt[ele] = 1
        
        for ele in mapt.keys():
            if ele in maps:
                if maps[ele] != mapt[ele]:
                    return ele
            else:
                return ele

File: 389-find-the-difference/find-the-difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        HashMap<Character, Integer> map1 = new HashMap<>();
        HashMap<Character, Integer> map2 = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (map1.containsKey(ch)) {
                int freq = map1.get(ch);
                map1.put(ch, freq + 1);
            } else {
                map1.put(ch, 1);
            }
        }

        for (int i = 0; i < t.length(); i++) {
            char ch = t.charAt(i);
            if (map2.containsKey(ch)) {
                int freq = map2.get(ch);
                map2.put(ch, freq + 1);
            } else {
                map2.put(ch, 1);
            }
        }

        for (char key : map2.keySet()) {
            if (!map1.containsKey(key))
                return key;
            if(map1.get(key) != map2.get(key))
                return key;
        }
        return 'a';
    }
}